### PR TITLE
Divide main.cpp

### PR DESCRIPTION
**Pull request by dl-dl**

### CODE REVIEW

**Review by larryk3 — CHANGES_REQUESTED** [review]
I locally tested the suggested changes, and it seems to work. But need to do more testing.

File: main/main.cpp [hunk 1]
```diff
@@ -269,37 +30,14 @@ extern "C" void app_main(void) {
...
-	while (!done)
-		;
-	delay(500);
-
-	// TODO figure out the memory stack required
-	const UBaseType_t priority = 30;
-	xTaskCreatePinnedToCore(taskAdcReadAndBuffer, "task_ADC_read", 1024 * 5, NULL, priority,
-	                        &adcReadTaskHandle, CORE_APP);
+	// Serial.printf("0x%" PRIx64 "\n", ESP.getEfuseMac());
+	Serial.println("temprary out of order\n");
```
[the remark this answers is not in the document]
Comment (dl-dl) [reply main/main.cpp:36]: I did not know what to include to make it compile.

[the remark this answers is not in the document]
Comment (larryk3) [reply main/main.cpp:36]: I think its arduino. The same function is called in the `ble_proc.cpp` file.
We can probably just call the underlying function directly and not depend on arduino as much.
```
uint64_t EspClass::getEfuseMac(void) {
  uint64_t _chipmacid = 0LL;
  esp_efuse_mac_get_default((uint8_t *)(&_chipmacid));
  return _chipmacid;
}
```

File: main/ble_proc.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+	// Start advertising
+	NimBLEAdvertising *pAdvertising = NimBLEDevice::getAdvertising();
+	pAdvertising->addServiceUUID(SERVICE_UUID);
+	// Standard BLE advertisement packet is only 31 bytes, so long names don't always fit.
+	// Scan response allows for devices to request more during the scan.
+	// This will allow for more than the 31 bytes, like longer names.
+	pAdvertising->setScanResponse(true);
+	NimBLEDevice::startAdvertising();
+}
```
[the remark this answers is not in the document]
Comment (dl-dl) [reply main/ble_proc.cpp]: Fixed.

**Review by dl-dl — COMMENTED** [review]
OK. Committed.

**Review by larryk3 — APPROVED** [review]

**Pull request merged**